pennylane/qchem/vibrational/taylor_ham.py

```diff
@@ -0,0 +1,550 @@
+        uloc (list(float)): localization matrix indicating the relationship between original and
+            localized modes
+
+    Returns:
+        BoseSentence: kinetic term of the taylor hamiltonian for given coeffs
+    """
+    taylor_1D = taylor_coeffs_array[0]
+    num_modes, _ = np.shape(taylor_1D)
+
+    if is_loc:
+        alphas_arr = np.einsum("ij,ik,j,k->jk", uloc, uloc, np.sqrt(freqs), np.sqrt(freqs))
+    else:
+        alphas_arr = np.zeros((num_modes, num_modes))
+        for m in range(num_modes):
+            alphas_arr[m, m] = freqs[m]
+
+    kin_ham = BoseSentence({})
+    for m1 in range(num_modes):
+        pm1 = _position_to_boson(m1, "p")
+        for m2 in range(num_modes):
+            pm2 = _position_to_boson(m2, "p")
+            kin_ham += (0.5 * alphas_arr[m1, m2]) * (pm1 * pm2).normal_order()
+
+    return kin_ham.normal_order()
+
+
+def _taylor_harmonic(taylor_coeffs_array, freqs):
+    """Build harmonic term of taylor form bosonic observable from provided coefficients, see first
+    term of `Eq. 4 and Eq. 7 <https://arxiv.org/pdf/1703.09313>`_.
+
+    Args:
+        taylor_coeffs_array (list(float)): the coeffs of the taylor expansion
+        freqs (list(float)): vibrational frequencies
+
+    Returns:
+        BoseSentence: harmonic term of the taylor hamiltonian for given coeffs
+    """
+    taylor_1D = taylor_coeffs_array[0]
+    num_modes, _ = np.shape(taylor_1D)
+    harm_pot = BoseSentence({})
+    for mode in range(num_modes):
+        bosonized_qm2 = (
+            _position_to_boson(mode, "q") * _position_to_boson(mode, "q")
+        ).normal_order()
+        harm_pot += bosonized_qm2 * freqs[mode] * 0.5
+
+    return harm_pot.normal_order()
+
+
+def taylor_bosonic(taylor_coeffs_array, freqs, is_loc=True, uloc=None):
+    """Build taylor form bosonic observable from provided coefficients, following `Eq. 4 and Eq. 7
+    <https://arxiv.org/pdf/1703.09313>`_.
+
+    Args:
+        taylor_coeffs_array (list(float)): the coeffs of the taylor expansion
+        freqs (list(float)): the harmonic frequencies in cm^-1
+        is_loc (bool): whether or not if localized
+        uloc (list(float)): localization matrix indicating the relationship between original and
+            localized modes
+
+    Returns:
+        BoseSentence: taylor hamiltonian for given coeffs
+    """
+    if is_loc:
+        start_deg = 2
+    else:
+        start_deg = 3
+
+    harm_pot = _taylor_harmonic(taylor_coeffs_array, freqs)
+    ham = _taylor_anharmonic(taylor_coeffs_array, start_deg) + harm_pot
+    kin_ham = _taylor_kinetic(taylor_coeffs_array, freqs, is_loc, uloc)
+    ham += kin_ham
+    return ham.normal_order()
+
+
+def taylor_hamiltonian(pes_object, deg=4, min_deg=3):
+    """Compute taylor hamiltonian from PES object
+
+    Args:
+        pes_object(VibrationalPES): the PES object
+        deg (int): the maximum degree of the taylor polynomial
+        min_deg (int): the minimum degree of the taylor polynomial
```

Comment:
Please update wrt the comments given above and make sure we are consistent with arg names.